child_compassion/model/child_compassion.py

```diff
@@ -476,25 +485,59 @@ def child_departed(self, cr, uid, ids, context=None):
         for child in self.browse(cr, uid, ids, context):
             if child.state == 'F':
                 child.write({'sponsor_id': False})
-                self.get_exit_details(cr, uid, child.id, context)
+                self.get_exit_details(cr, uid, child.id, context)
         return True
 
-    def child_remove_from_typo3(self, cr, uid, ids, context=None):
-        child_codes = [child.code for child in self.browse(cr, uid, ids,
-                                                           context)]
-        filename = "upd.sql"
+    def _get_typo3_child_id(self, cr, uid, child_code, context=None):
+        res = self._sel_request_to_typo3(
+            cr, uid,
+            "select * "
+            "from tx_drechildpoolmanagement_domain_model_children "
+            "where child_key='%s'" % child_code,
+            context)
+
+        return json.loads(res)[0]['uid']
+
+    def _sel_request_to_typo3(self, cr, uid, request, context=None):

```

Comment:
`_sel_request_to_typo3` and `_update_request_to_typo3` are almost the same method except for file_name used and script_name called. You could merge these two methods and add parameters for these different values to save lines of code.